packages/amplication-code-gen-types/src/code-gen-types.ts

```diff
@@ -129,6 +124,76 @@ export type Module = {
   code: string;
 };
 
+/**
+ * ModuleMap is a map of module paths to modules
+ */
+export class ModuleMap {
+  private map: Record<string, Module> = {};
+  constructor(private readonly logger: BuildLogger) {}
+
+  async merge(anotherMap: ModuleMap): Promise<void> {
+    for await (const module of anotherMap.modules()) {
+      await this.set(module.path, module);
+    }
+  }
+
+  async mergeMany(maps: ModuleMap[]): Promise<void> {
+    for await (const map of maps) {
+      await this.merge(map);
+    }
+  }
+
+  async set(path: string, module: Module) {
+    if (this.map[path]) {
+      await this.logger.warn(`Module ${path} already exists. Overwriting...`);
+    }
+    this.map[path] = module;
+  }
```

Comment:
Auto review for @overbit. How about changing the signature of set to accept only `Module`?

```suggestion
  async set(module: Module) {
    if (this.map[module.path]) {
      await this.logger.warn(`Module ${module.path} already exists. Overwriting...`);
    }
    this.map[module.path] = module;
  }
```

Reply:
@overbit I Agree. let's use only module